File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/utils/StringUtils.java
package cn.tongdun.mobrisk.core.utils;

import java.util.ArrayList;
import java.util.List;

/**
 * @description: string util
 * @author: wuzuchang
 * @date: 2022/12/6
 */
public class StringUtils {

    public static boolean isEmpty(String str) {
        return ((str == null) || ("".equals(str.trim()) || str.length() == 0));
    }

    public static List<String> splitNonRegex(String input, String delim) {
        List<String> list = new ArrayList<>();
        if (input == null || input.length() == 0) {
            return list;
        }
        if (delim == null || delim.length() == 0) {
            list.add(input);
            return list;
        }
        if(input.equals(delim)){
            return list;
        }
        while (true) {
            int index;
            index = input.indexOf(delim);
            if (index == -1) {
                if (!isEmpty(input)) {
                    list.add(input);
                }
                return list;
            }
            if (index != 0) {
                list.add(input.substring(0, index));
            }
            input = input.substring(index + delim.length());
        }
    }

    public static String byteToHexString(byte[] data){
        StringBuilder hex = new StringBuilder();
        // Iterating through each byte in the array
        for (byte i : data) {
            hex.append(String.format("%02X", i));
        }
        return hex.toString();
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/tools/StringUtils.kt
package cn.tongdun.mobrisk.core.tools

import java.nio.charset.StandardCharsets
import java.security.MessageDigest

/**
 * @description:
 * @author: wuzuchang
 * @date: 2023/5/15
 */

fun String.hash(algorithm: String): String {
    return executeSafe({
        val md = MessageDigest.getInstance(algorithm)
        md.update(this.toByteArray(StandardCharsets.UTF_8))
        md.digest().toUppercaseHexString()
    }, "")
}

fun String.splitNonRegex(delimiter: String): List<String> {
    val list: MutableList<String> = mutableListOf()
    var input = this
    if (input.isEmpty()) {
        return list
    }
    if (delimiter.isEmpty()) {
        list.add(input)
        return list
    }
    if (input == delimiter) {
        return list
    }
    while (true) {
        val index: Int = input.indexOf(delimiter)
        if (index == -1) {
            if (input.isNotEmpty()) {
                list.add(input)
            }
            return list
        }
        if (index != 0) {
            list.add(input.substring(0, index))
        }
        input = input.substring(index + delimiter.length)
    }
}

fun ByteArray.hash(algorithm: String): String {
    return executeSafe({
        val md = MessageDigest.getInstance(algorithm)
        md.update(this)
        md.digest().toUppercaseHexString()
    }, "")
}

fun ByteArray.toUppercaseHexString(): String {
    val hex = StringBuilder()
    // Iterating through each byte in the array
    for (i in this) {
        hex.append(String.format("%02X", i))
    }
    return hex.toString()
}